packages/office-ui-fabric-react/src/components/FocusZone/FocusZone.tsx

```diff
@@ -221,6 +221,7 @@ export class FocusZone extends BaseComponent<IFocusZoneProps, {}> implements IFo
       while (parentElement && parentElement !== this._root.value) {
         if (isElementTabbable(parentElement) && this._isImmediateDescendantOfZone(parentElement)) {
           this._activeElement = parentElement;
+          this._setFocusAlignment(this._activeElement);
```

Comment:
I think this will cause focus to be "pulled" to one side for variable sized boxes.

Reply:
I don't think so. The code at lines 216-217 run when the event target is the immediate descendent of the parent. If it's not, this while loop just finds the parent element that is an immediate descendent, and should therefore be running the same code once it finds that parent element. Looks like a bug that we were only updating focus alignment if the focused element was a top level element.

The case where focus gets pulled to the side was caused by my changes down in _setFocusAlignment(), e.g. we only want to update the horizontal alignment when going left or right, the onFocus shouldn't update the alignment except for the case where no alignment was previously set.